ui/appui-react/src/appui-react/widget-panels/ToolSettings.tsx

```diff
@@ -74,17 +74,23 @@ export function WidgetPanelsToolSettings() {
 /** @internal */
 export function ToolSettingsDockedContent() {
   const settings = useHorizontalToolSettingNodes();
+  const [forceRefreshKey, setForceRefreshKey] = React.useState(Date.now());
+  React.useEffect(() => {
+    // We cant work with the content of the settings, but we can force refresh when
+    // the array is different.
+    setForceRefreshKey(Date.now());
+  }, [settings]);
   // for the overflow to work properly each setting in the DockedToolSettings should be wrapped by a DockedToolSetting component
   return (
     <DockedToolSettings
       itemId={
         InternalFrontstageManager.activeToolSettingsProvider?.uniqueId ?? "none"
       }
-      key={Date.now()}
+      key={forceRefreshKey}
```

Comment:
Can we just get rid of the key or do we really need to re-mount the component?

Reply:
We need to remount the component because we are only placing pre rendered component in there, and since they are only indexed by key underneath, if the node changes they are not rerendered. So the content of the node change, if the event that change them is known by the node (by a Hook for example), but if the whole node is changed, nothing happens because they all have the same key, and since we dont have any insight of what is in that node, there isnt much we can do about it.

Maybe we could redesign the whole system, and build a key out of the toolsettings values/editor, but that would likely break more stuf that fix, and require dev updating their stuff, so I went with this as we have 2 people waiting on this.